### PR TITLE
Add collection support to operation

### PR DESCRIPTION
**Pull request by fernandes**
same as #39 but with squashed commits

### CODE REVIEW

File: test/crud_test.rb [hunk 1]
```diff
@@ -101,6 +106,45 @@ class ModelUpdateOperation < CreateOperation
...
+  # model Song, :action
+  class FetchCollectionOperation < CreateOperation
+    model Song
+
+    contract do
+      property :title
+    end
+
+    def setup_model!(params)
+      if @user_role == 'admin'
```
Comment (apotonick) [inline test/crud_test.rb]: Sidenote: I don't think we need this differentiation between different roles here. It's good to see how you solve that problem, but a bit "off-topic" for this test, as we really only test fetching a collection here.

File: test/crud_test.rb [hunk 2]
```diff
@@ -101,6 +106,45 @@ class ModelUpdateOperation < CreateOperation
...
+        @collection = Song.all
+      else
+        @collection = nil
+      end
+    end
+
+    def setup_params!(params)
+      @user_role = "admin" if params[:user_id] == 0
```
Comment (fernandes) [inline test/crud_test.rb]: I'm setting @user_role here, idea is just differentiate on how fetch two different collection, could be:

```ruby
@myfoovar = true if params[:myfoovar] == 1
```

then if @myfoovar above

File: test/crud_test.rb (hunk 1 above)
Comment (fernandes) [inline test/crud_test.rb]: just to show an example on how differentiate and on fetching collections, you can change to whatever var name you want hehehhe

**Comment by apotonick** [comment]
:kiss:

**Comment by fernandes** [comment]
iahuhulllllll :kissing:

**Pull request merged**